syft/federated/plan.py

```diff
@@ -20,41 +21,53 @@ def make_plan(plan_blueprint):
     return func2plan(plan_blueprint)
 
 
-def func2plan(plan_blueprint):
+class func2plan(object):
     """Converts a function to a plan.
 
     Converts a function containing sequential pytorch code into
     a plan object which can be sent to any arbitrary worker.
+
+    This class should be used only as decorator.
     """
-    plan = Plan(owner=sy.local_worker, id=sy.ID_PROVIDER.pop(), name=plan_blueprint.__name__)
-    plan.blueprint = plan_blueprint
-    return plan
+
+    def __init__(self, args_shape=None):
+        self.args_shape = args_shape
+
+    def __call__(self, plan_blueprint):
```

Comment:
Rather than building plans at call time, could we build them at send time?

Reply:
That's how is currently done, but it will not work for remote workers (aka WebSocket workers)

Reply:
What in particular makes them break at send time for websocket workers?

Reply:
The main thing is that virtual workers are able to build a plan remotely because they can run the model blueprint (original function / method) this is not true for remote workers.

Reply:
The proxy of the websocket worker is not doing anything at the moment. Just transmitting the message to the remote counterpart. And the remote part does not have access to the blueprint.
So if the proxy got intelligent and built the plan before sending it, it could work.

Reply:
@robert-wagner @midokura-silvia Actually, I've just noticed what I've said previously was not true.

> That's how is currently done, but it will not work for remote workers (aka WebSocket workers)

[An unbuilt plan can be sent successfully to a remote worker](https://gist.github.com/mari-linhares/d1feeb45903dc15c76dde1c5874b69c8) (if you add this test to `test_plan.py` in dev branch it works).

[But the current version can't be executed locally successfully](https://gist.github.com/mari-linhares/a30ea03066a9dc5c45c613624595447b). This PR solves this.

Reply:
@LaRiffle do you think is good that we force the model to be built before being sent or since this is not particularly a problem we shouldn't?

Reply:
I think so yes, otherwise we will use the similar tricks as before with lazy .send() which don't really send the plan until a call is made. This, which was previously used, is complicated to understand, use and debug. Does it make sense?

Reply:
Yes, I agree.